### PR TITLE
Adaptive QP scaling

### LINKED ISSUE #294: Content Adaptive QP Scaling
Description: add content adaptive QP Scaling assignment to non-key frames 
Trello reference: https://trello.com/c/IOQiZ1Jy/58-content-adaptive-qp-scaling
Current working branch: https://github.com/hguermaz/SVT-AV1/tree/QP_SCALING

### PR DESCRIPTION
**Pull request by anaghdin**
## Description
Change the QP based on the content. Adaptive QP scaling was previously active in I slices only.
Updated the algorithm to use a new metric for all frames. The metric is defined as normalized ME distortion (16x16 based from list 0) over frames in the look ahead.
If number of frames in the LAD/SW is less than 8, the adaptive QP scaling algorithm is turned off.

Associated with Issue #294
Closes #294

## Authors

@anaghdin

## Type of change

Feature improvement

## Tests and performance

Expected Average PSNR-SSIM BD-rate gain in the range of -2.4% on the AOM test set, using 4 QP values: {20, 32, 43 and 55}. Note that the maximum gain is achieved when ALT_REF is enabled

No Speed impact.

## TODO items:

The algorithm can be further improved using other information from LAD or ALR_REF filtering

### CODE REVIEW

File: Source/Lib/Common/Codec/EbMotionEstimationProcess.c [hunk 1]
```diff
@@ -493,8 +494,21 @@ EbErrorType ComputeDecimatedZzSad(
...
+            else if (decimatedLcuCollocatedSad < ((decimatedLcuWidth * decimatedLcuHeight) * 4)) {
+                previous_picture_control_set_wrapper_ptr->non_moving_index_array[sb_index] = BEA_CLASS_1_ZZ_COST;
+            }
+            else if (decimatedLcuCollocatedSad < ((decimatedLcuWidth * decimatedLcuHeight) * 8)) {
+                previous_picture_control_set_wrapper_ptr->non_moving_index_array[sb_index] = BEA_CLASS_2_ZZ_COST;
+            }
+            else {
+                previous_picture_control_set_wrapper_ptr->non_moving_index_array[sb_index] = BEA_CLASS_3_ZZ_COST;
+            }
+#else
```
Comment (joelsole) [inline Source/Lib/Common/Codec/EbMotionEstimationProcess.c]: no braces

Comment (anaghdin) [reply Source/Lib/Common/Codec/EbMotionEstimationProcess.c]: Addressed

**Comment by anaghdin** [comment]
> There are warnings on Travis.

AN: Thanks for pointing out. Addressed

**Comment by 1480c1** [comment]
@hassount or @anaghdin / any other maintainers, please clean up the branches after they are merged and after you are sure there will be no more development on those branches, there are already 18 branches and it will be harder in the future to keep the list clean

**Pull request merged**